### PR TITLE
Merge railgun and kubernetes-testing-framework into KIC

### PR DESCRIPTION
**Pull request by mflendrich**
- Copies commits from the railgun repository into `railgun/`.
- Copies commits from the `kubernetes-testing-framework` repository into `pkg/k8stest`.
- Unifies the `railgun` Go module with the KIC Go module (which means adding and upgrading several dependencies of KIC).
- Unifies the `pkg/k8stest` Go module with the KIC Go module (because doing otherwise meant `go mod` trying to resolve from other branches).
- Fixes a missing build tag in the integration test `TestMain()` file.

### CODE REVIEW

File: railgun/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Kubernetes Ingress Controller (KIC) Prototype
+
+This repo holds a prototype implementation of the [Kong Kubernetes Ingress Controller (KIC)][kic] updated and built for [Kubebuilder][kb] and [Controller Runtime v0.7.x+][ctrl].
+
+The naming `railgun` is a codename used arbitrarily based on a [video game reference][q3].
+
+If you're confused what you're looking at or have any questions, please reach out to @kong/team-k8s on our [Slack Channel][slack]!
```
Comment (shaneutt) [inline railgun/README.md:7]:
```suggestion
```

File: railgun/README.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+```shell
+export GOPRIVATE="github.com/kong/kubernetes-testing-framework,${GOPRIVATE}"
+git config --global url."git@github.com:kong/kubernetes-testing-framework".insteadOf https://github.com/kong/kubernetes-testing-framework
+```
+
+[kic]:https://github.com/kong/kubernetes-ingress-controller
+[kb]:https://github.com/kubernetes-sigs/kubebuilder
+[ctrl]:https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.7.0
+[q3]:https://github.com/ioquake/ioq3
+[slack]:https://app.slack.com/client/T0DS5NB27/C011RQPHDC7
```
Comment (shaneutt) [inline railgun/README.md:58]:
```suggestion
```

File: railgun/README.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+# Tests
+
+Right now the integration tests in `test/integration` rely on a private repo, make sure you run the following before running tests:
+
+```shell
+export GOPRIVATE="github.com/kong/kubernetes-testing-framework,${GOPRIVATE}"
+git config --global url."git@github.com:kong/kubernetes-testing-framework".insteadOf https://github.com/kong/kubernetes-testing-framework
+```
+
```
Comment (shaneutt) [inline railgun/README.md:53]:
```suggestion
```

**Review by shaneutt — DISMISSED** [review]
We have a couple README/docs issues and then LGTM

**Review by shaneutt — APPROVED** [review]
🎉

**Comment by mflendrich** [comment]
Merge commit  857d942 deleted from `next` due to issues with commit attribution. Consider this PR effectively unmerged.

It appears that GitHub refuses to list authors of subtree commits as file contributors, which is unfair for them.

**Pull request merged**